### PR TITLE
Random runner updates

### LINKED ISSUE #574: [NEW FEATURE] output mask in healpix format
Footprint masks in healpix format are very useful.

The random catalogue runner can be extended to output the pipeline flag mask file in healpix format.

### PR DESCRIPTION
**Pull request by martinkilbinger**
## Summary

- Added options to output mask as healpix FITS image.
- Removed reading and use of list of all tiles for overlapping region detection

Closes #574 

## Reviewer Checklist

> Reviewers should tick the following boxes before approving and merging the PR.

- [x] The PR targets the `develop` branch
- [x] The PR is assigned to the developer
- [x] The PR has appropriate labels
- [x] The PR is included in appropriate projects and/or milestones
- [x] The PR includes a clear description of the proposed changes
- [x] If the PR addresses an open issue the description includes "closes #<ISSUE NUMBER>"
- [x] The code and documentation style match the current standards
- [x] Documentation has been added/updated consistently with the code
- [x] All CI tests are passing
- [x] API docs have been built and checked at least once (if relevant)
- [x] All changed files have been checked and comments provided to the developer
- [x] All of the reviewer's comments have been satisfactorily addressed by the developer

### CODE REVIEW

File: shapepipe/modules/random_cat_package/random_cat.py [hunk 1]
```diff
@@ -29,33 +33,79 @@ class RandomCat():
...
+    def save_as_healpix(self, hdu_mask, header):
+        """Save As Healpix.
+
+        Save mask as healpix FITS file.
+
+        Parameters
+        ----------
+        hdu_mask : HDU list
+            HDU with 2D pixel mask image
+        header : FITS header
```
Comment (sfarrens) [inline shapepipe/modules/random_cat_package/random_cat.py:84]: These don't look like real data types

File: shapepipe/modules/random_cat_package/random_cat.py [hunk 2]
```diff
@@ -155,10 +208,15 @@ def process(self):
...
         # Tile ID
+        output_path = (
+            f'{output_dir}/{output_file_pattern}-'
+            + f'{file_number_string}.fits'
+        )
         file_name = os.path.split(self._output_path)[1]
         file_base = os.path.splitext(file_name)[0]
         tile_ID_str = re.split('-', file_base)[1:]
         tile_id = float('.'.join(tile_ID_str))
+        print('MKDEBUG', tile_id, file_number_string)
```
Comment (sfarrens) [inline shapepipe/modules/random_cat_package/random_cat.py:219]: I guess this is to be removed

File: shapepipe/modules/random_cat_package/random_cat.py [hunk 3]
```diff
@@ -29,33 +33,79 @@ class RandomCat():
...
+        Output file pattern (base name) for random catalogue
     n_rand : float
         Number of random objects on output
     density : bool
         ``n_rand`` is interpreted per square degrees if ``True``
     w_log : logging.Logger
         Logging instance
-
+    healpix_options : dict
+        Parameters for healpix output mask file
```
Comment (sfarrens) [inline shapepipe/modules/random_cat_package/random_cat.py:49]:
```suggestion
        Parameters for HEALPix output mask file
```

File: shapepipe/modules/random_cat_package/random_cat.py [hunk 4]
```diff
@@ -29,33 +33,79 @@ class RandomCat():
...
+        """Save As Healpix.
+
+        Save mask as healpix FITS file.
+
+        Parameters
+        ----------
+        hdu_mask : HDU list
+            HDU with 2D pixel mask image
+        header : FITS header
+            image header with WCS information
```
Comment (sfarrens) [inline shapepipe/modules/random_cat_package/random_cat.py:85]:
```suggestion
            Image header with WCS information
```

**Review by sfarrens — CHANGES_REQUESTED** [review]

File: shapepipe/modules/random_cat_package/random_cat.py (hunk 1 above)
Comment (martinkilbinger) [reply shapepipe/modules/random_cat_package/random_cat.py]: how about now?

File: shapepipe/modules/random_cat_package/random_cat.py (hunk 2 above)
Comment (martinkilbinger) [reply shapepipe/modules/random_cat_package/random_cat.py]: removed

**Comment by sfarrens** [comment]
Hey @martinkilbinger can you just need to fix these minor formatting errors, then the CI tests should pass.

```
______________________________ pycodestyle-check _______________________________
/home/runner/work/shapepipe/shapepipe/shapepipe/modules/mask_package/mask.py:237:12: E275 missing whitespace after keyword

______________________________ pycodestyle-check _______________________________
/home/runner/work/shapepipe/shapepipe/shapepipe/modules/setools_package/setools.py:656:18: E275 missing whitespace after keyword

______________________________ pycodestyle-check _______________________________
/home/runner/work/shapepipe/shapepipe/shapepipe/pipeline/file_io.py:554:24: E275 missing whitespace after keyword
```

**Comment by martinkilbinger** [comment]
> Hey @martinkilbinger can you just need to fix these minor formatting errors, then the CI tests should pass.
> 
> ```
> ______________________________ pycodestyle-check _______________________________
> /home/runner/work/shapepipe/shapepipe/shapepipe/modules/mask_package/mask.py:237:12: E275 missing whitespace after keyword
> 
> ______________________________ pycodestyle-check _______________________________
> /home/runner/work/shapepipe/shapepipe/shapepipe/modules/setools_package/setools.py:656:18: E275 missing whitespace after keyword
> 
> ______________________________ pycodestyle-check _______________________________
> /home/runner/work/shapepipe/shapepipe/shapepipe/pipeline/file_io.py:554:24: E275 missing whitespace after keyword
> ```

Done, it should pass all tests now.

**Pull request merged**